src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -216,6 +218,22 @@ private void SetGlobalErrorIfTrue(bool arg)
             }
         }
 
+        private static void AddCircularStructDiagnostics(NamespaceOrTypeSymbol symbol)
+        {
+            if (symbol is SourceMemberContainerTypeSymbol sourceMemberContainerTypeSymbol)
+            {
+                _ = sourceMemberContainerTypeSymbol.KnownCircularStruct;
+            }
+
+            foreach (var member in symbol.GetMembersUnordered())
+            {
+                if (member is NamespaceOrTypeSymbol namespaceOrTypeSymbol)
+                {
+                    AddCircularStructDiagnostics(namespaceOrTypeSymbol);
```

Comment:
>AddCircularStructDiagnostics [](http://example.com/codeflow?start=20&length=28)

It looks like we are not parallelizing this operation. See VisitNamespace/VisitNamedType, for example.

Reply:
@AlekseyTs Done in last commit. I'm not sure if I did it correctly?